projects/solace-message-client/src/lib/solace-message-client.ts

```diff
@@ -218,7 +218,12 @@ export abstract class SolaceMessageClient {
    *         {@link Message} object instead, which you can construct using {@link SolclientFactory#createMessage}.
    *         For more information, refer to the documentation of {@link Data}.
    * @param  options - Controls how to publish the message.
-   * @return A Promise that resolves when dispatched the message, or that rejects if the message could not be dispatched.
+   * @return A Promise that resolves when the message could be successfully dispatched or that rejects if the message could not be dispatched.
+   *         A Promise is resolved immediately if the published message has a delivery mode {@link MessageDeliveryModeType#DIRECT} (direct messaging).
+   *         If the published message has a delivery mode {@link MessageDeliveryModeType#PERSISTENT} (guaranteed or persistent messaging),
+   *         the acknowledgment mechanism should be considered which is described here: https://docs.solace.com/API/API-Developer-Guide/Acknowledging-Published-.htm
+   *         That means, the promise is resolved when a threshold of either a third of the Publish Window Size is reached or one second has elapsed.
+   *         This is a default behavior how the event broker sends an acknowledgment for published messages.
```

Comment:
```suggestion
   * @return A Promise that resolves when the message could be successfully dispatched or that rejects otherwise.
   *         The Promise resolves immediately if the published message has the delivery mode {@link MessageDeliveryModeType#DIRECT} (direct messaging).
   *         If the published message has the delivery mode {@link MessageDeliveryModeType#PERSISTENT} (guaranteed or persistent messaging),
   *         the Promise resolves once acknowledged, i,e., when one-third of the publish window size is reached or one second has elapsed. Refer to https://docs.solace.com/API/API-Developer-Guide/Acknowledging-Published-.htm for more information.
```

Reply:
Great, thanks for the more precise wording. I changed the explanation a bit. What do you think about it?

Reply:
Looks fine for more. I took over your changes.